tests/test_mariadb.py

```diff
@@ -276,3 +281,80 @@ def test_mariadb_healthcheck_galera_cluster_disabled(auto_container_per_test):
     _wait_for_server(conn)
 
     conn.run_expect([1], "healthcheck.sh --su-mysql --galera_online")
+
+
+_DB_ENV = {
+    "MARIADB_USER": _OTHER_DB_USER,
+    "MARIADB_PASSWORD": _OTHER_DB_PW,
+    "MARIADB_DATABASE": _TEST_DB,
+    "MARIADB_ROOT_PASSWORD": MARIADB_ROOT_PASSWORD,
+    "MARIADB_AUTO_UPGRADE": "1",
+}
+
+OS_VERSION = os.getenv("OS_VERSION", "")
+
+
+@pytest.mark.parametrize("ctr_image", MARIADB_CONTAINERS)
+@pytest.mark.skipif(
+    OS_VERSION in ("15.5",), reason="MariaDB upgrade scenario not supported"
+)
+def test_mariadb_upgrade(
+    container_runtime: OciRuntimeBase,
+    pytestconfig: pytest.Config,
+    ctr_image: DerivedContainer,
+    tmp_path: Path,
+) -> None:
+    mounts = [BindMount(host_path=tmp_path, container_path="/var/lib/mysql")]
+    mariadb_old = Container(
+        url="registry.opensuse.org/devel/bci/sle-15-sp6/containerfile/suse/mariadb:latest",
+        volume_mounts=mounts,
+        extra_environment_variables=_DB_ENV,
+    )
+    mariadb_new = DerivedContainer(
+        base=ctr_image,
+        volume_mounts=mounts,
+        extra_environment_variables=_DB_ENV,
+    )
+    with ContainerLauncher.from_pytestconfig(
+        mariadb_old, container_runtime, pytestconfig
+    ) as launcher:
+        launcher.launch_container()
+        con = launcher.container_data.connection
+        _wait_for_server(con)
+        mariadb_cmd = f"mariadb --user={_OTHER_DB_USER} --password={_OTHER_DB_PW} --host=0.0.0.0 {_TEST_DB}"
+
+        con.check_output(
+            f'echo "CREATE TABLE test (id serial PRIMARY KEY, num integer, data varchar(32));" | {mariadb_cmd}'
+        )
+        con.check_output(
+            f"echo 'INSERT INTO test (num, data) VALUES (100, \"abcdef\")' | {mariadb_cmd}"
+        )
+        rows = (
+            con.check_output(f'echo "SELECT * FROM test;" | {mariadb_cmd}')
+            .strip()
+            .splitlines()
+        )
+
+        assert rows and len(rows) == 2
+        _, num, data = rows[-1].split()
+        assert num == "100" and data == "abcdef"
+
+    with ContainerLauncher.from_pytestconfig(
+        mariadb_new, container_runtime, pytestconfig
+    ) as launcher:
+        launcher.launch_container()
+        con = launcher.container_data.connection
+        _wait_for_server(con)
+        expected_version = mariadb_new.url.split(":")[1]
+        if expected_version != "latest":
+            assert expected_version in con.check_output("mysql --version")
```

Comment:
I don't get the point of this check to be honest

Reply:
this is just an additional check that the upgraded is done successfully. But, yes it is not mandatory.
It is expected that the mariadb image:x.y to run Mariadb-Vx.y.

Reply:
we can ignore this check.

Reply:
I don't understand what a version check has to do with the upgrade?

Reply:
this is just a check that our database is upgraded to newer version.

Reply:
removed this section as it doesn't make much sense.